Makefile

```diff
@@ -89,14 +89,31 @@ deps\:chglog:
 deps\:modules:
 	go mod download
 
+.PHONY: deps\:mock
+deps\:mock:
+	go install github.com/vektra/mockery/v2@v2.30.1
+
 .PHONY: deps
 deps:
 	@$(MAKE) deps:modules && \
 	$(MAKE) deps:bench && \
 	$(MAKE) deps:chglog && \
 	$(MAKE) deps:coverage && \
 	$(MAKE) deps:lint && \
-	$(MAKE) deps:test
+	$(MAKE) deps:test && \
+	$(MAKE) deps:mock
+
+.PHONY: mock
+mock:
+	$(MAKE) deps:mock
+	mockery --dir ./client --output ./client/mocks --name DB --with-expecter
+	mockery --dir ./client --output ./client/mocks --name Collection --with-expecter
+	mockery --dir ./datastore --output ./datastore/mocks --name DAGStore --with-expecter
+	mockery --dir ./datastore --output ./datastore/mocks --name DSReaderWriter --with-expecter
+	mockery --srcpkg github.com/ipfs/go-datastore/query --output ./datastore/mocks --name Results --with-expecter
+	mockery --dir ./datastore --output ./datastore/mocks --name RootStore --with-expecter
+	mockery --dir ./datastore --output ./datastore/mocks --name Txn --with-expecter
+	mockery --dir ./datastore --output ./datastore/mocks --name DAGStore --with-expecter
```

Comment:
Would we want to force `with-expecter: True` for everything though? Are there any downsides besides a few extra auto-generated lines?

RE the bash, for now at least I don't think such a script would improve the readability of the codebase.  I'd much rather have a few extra lines.

Reply:
thought: don't need a bash script I think might be overkill.

thought: Could we do all these in a mockery config file, like define all the packages there and a global `with-expecter: True` and the output dir and everything there?

Example from their documentation:
```
with-expecter: True
packages:
  github.com/vektra/mockery/v2/pkg: # (1)!
    interfaces:
      TypesPackage:
      RequesterVariadic:
        config: # (2)!
          with-expecter: False 
        configs:
          - mockname: MockRequesterVariadicOneArgument
            unroll-variadic: False
          - mockname: MockRequesterVariadic
  io:
    config:
      all: True # (3)!
    interfaces:
      Writer:
        config:
          with-expecter: False # (4)!
```

You can privide these as config args in the file:

```
| name | templated | default | description |
|------|-----------|---------|-------------|
| `all`  |  :fontawesome-solid-x: | `#!yaml false` | Generate all interfaces for the specified packages. |
| `boilerplate-file` | :fontawesome-solid-x: | `#!yaml ""` | Specify a path to a file that contains comments you want displayed at the top of all generated mock files. This is commonly used to display license headers at the top of your source code. |
| `config` | :fontawesome-solid-x: | `#!yaml ""` | Set the location of the mockery config file. |
| `dir` | :fontawesome-solid-check: | `#!yaml "mocks/{{.PackagePath}}"` | The directory where the mock file will be outputted to. |
| `disable-config-search` | :fontawesome-solid-x: | `#!yaml false` | Disable searching for configuration files |
| `disable-version-string` | :fontawesome-solid-x: | `#!yaml false` | Disable the version string in the generated mock files. |
| `dry-run` | :fontawesome-solid-x: | `#!yaml false` | Print the actions that would be taken, but don't perform the actions. |
| `filename` | :fontawesome-solid-check: | `#!yaml "mock_{{.InterfaceName}}.go"` | The name of the file the mock will reside in. |
| `inpackage` | :fontawesome-solid-x: | `#!yaml false` | When generating mocks alongside the original interfaces, you must specify `inpackage: True` to inform mockery that the mock is being placed in the same package as the original interface. |
| `mockname` | :fontawesome-solid-check: | `#!yaml "Mock{{.InterfaceName}}"` | The name of the generated mock. |
| `outpkg` | :fontawesome-solid-check: | `#!yaml "{{.PackageName}}"` | Use `outpkg` to specify the package name of the generated mocks. |
| `log-level` | :fontawesome-solid-x: | `#!yaml "info"` | Set the level of the logger |
| [`packages`](features.md#packages-configuration) | :fontawesome-solid-x: | `#!yaml null` | A dictionary containing configuration describing the packages and interfaces to generate mocks for. |
| `print` | :fontawesome-solid-x: | `#!yaml false` | Use `print: True` to have the resulting code printed out instead of written to disk. |
| [`recursive`](features.md#recursive-package-discovery) | :fontawesome-solid-x: | `#!yaml false` | When set to `true` on a particular package, mockery will recursively search for all sub-packages and inject those packages into the config map. |
| `tags` | :fontawesome-solid-x: | `#!yaml ""` | Set the build tags of the generated mocks. |
| [`with-expecter`](features.md#expecter-structs) | :fontawesome-solid-x: | `#!yaml true` | Use `with-expecter: True` to generate `EXPECT()` methods for your mocks. This is the preferred way to setup your mocks. |
| [`replace-type`](features.md#replace-types) | :fontawesome-solid-x: | `#!yaml null` | Replaces aliases, packages and/or types during generation.|
```

Links:
https://github.com/vektra/mockery/blob/master/docs/configuration.md
https://github.com/vektra/mockery/blob/master/docs/features.md

Reply:
> thought: Could we do all these in a mockery config file, like define all the packages there and a global with-expecter: True and the output dir and everything there

Similar to the bash-script, I dont think that is worth it right now and might have a negative impact on readability.  I also like how simple and easy the individual commands are to read and copy-paste if someone just wants to regenerate a single (or take as a template to generate a new one)